### DISCUSSION TITLE
Why close the fiber ID in the fiber itself and then in the main thread?

### DISCUSSION THREAD

**Discussion by vovkapoc** [revoltphp/event-loop#92] [Q&A] [answered]
From documentation ( https://revolt.run/fundamentals ):
```
$readableId = EventLoop::onReadable(STDIN, function ($id, $stream) use ($suspension): void {
    EventLoop::cancel($id);
    // ... other code ...
});

// ... other code
EventLoop::cancel($readableId);
```

I test this, $readableId and $id in anonymous callback is are the same. Please explain why they need to be closed both in the middle of the callback and in the main thread, although the fiber ID is the same both there and there...

**_Sorry for my bad English, I from Ukraine_**

**Comment by kelunik** [comment, marked as answer]
There are two such calls, because the one inside the callback cancels further invocations after the first invocation. The other one cancels the readability callback in case the timeout is reached.

**Reply by vovkapoc** [reply]
Thank you very much!😎👍
